files/galaxy/tpv/destinations.yml

```diff
@@ -28,8 +27,8 @@ destinations:
       submit_request_cpus: "{cores}"
       submit_request_memory: "{mem}G"
       singularity_enabled: true
-      singularity_volumes: "$_CONDOR_SCRATCH_DIR:rw,$job_directory:rw,$tool_directory:ro,$job_directory/outputs:rw,$working_directory:rw,/data/db/:ro,/data/dnb01/galaxy_db/:ro,/data/dnb02/galaxy_db/:ro,/data/dnb03/galaxy_db/:ro,/data/dnb05/galaxy_db/:ro,/data/dnb06/galaxy_db/:rw,/data/dnb07/galaxy_db/:rw,/data/dnb08/galaxy_db/:rw,/data/dp01/galaxy_db/:rw,/data/0/galaxy_db/:ro,/data/1/galaxy_db/:ro,/data/2/galaxy_db/:ro,/data/3/galaxy_db/:ro,/data/4/galaxy_db/:ro,/data/5/galaxy_import/galaxy_user_data/:ro,/data/6/galaxy_db/:ro,/data/7/galaxy_db/:ro,/usr/local/tools/:ro"
-      singularity_default_container_id: "/cvmfs/singularity.galaxyproject.org/all/centos:8.3.2011"
+      singularity_volumes: "$_CONDOR_SCRATCH_DIR:rw,$job_directory:rw,$tool_directory:ro,$job_directory/outputs:rw,$working_directory:rw,{% for vol in dnb.values() %}{{ vol.path }}/galaxy_db/:{{ vol.docker_perm }},{% endfor %}{{ tools.tools.path }}:{{ tools.tools.docker_perm }}"
```

Comment:
If we switch to templates, then we need then the same trick required for setting up the TPV dry-run in Github actions (running Ansible just to fill the templates) to get the TPV linter to work again...

It's a pity that Jenkins does not do this honestly.
![grafik](https://github.com/usegalaxy-eu/infrastructure-playbook/assets/43052541/fc6f62cb-3f65-4766-be11-86849c8584d0)

Reply:
This way we define the string just once and reuse it everywhere.

Reply:
But we are using stuff here (i.e. `for vol in dnb.values()`) that also needs to be shared with Ansible group variables, right? Defining a TPV variable somewhere may imply templating that file, and then we would run into the same problem.

Reply:
For other use cases where things are only applied to TPV, I assume the answer is yes, at worst as an environment variable or similar.

However, I expect the solution to be more verbose than using Jinja templates (we could always construct that string with Jinja templates).

Reply:
Assuming the TPV var would be reused a few times, that would work well :+1:

Reply:
I am now wondering though: aren't we doing premature optimization here? Such a variable would only be reused here and in the basic docker destination right?

Reply:
Maybe, it was just a suggestion/Q.
I thought the volume string on its own, will be shared between all Docker and Singularity destinations etc ...

Reply:
I think we can manage that with inheritance

Reply:
> I think we can manage that with inheritance

Exactly I think this is already working well.